### DISCUSSION TITLE
Old app behind new proxy, ajax and non-ajax URLs revert to old URL with server name instead of proxy prefix

### DISCUSSION THREAD

**Discussion by arnieHouston** [primefaces/community#2648] [PrimeFaces] [answered]
Is there some technique to deal with this? Some of the URLs in this app are VERY relative--as in the same folder of xhtml files. Command links or buttons exhibit random behavior where there's a pause then the URL bar shows the server name:port where the app lives instead of the proxy server URL prefix.

**Comment by melloware** [comment]
This sounds more like a general Java Servlet EE JSF question than a PF question?

**Comment by arnieHouston** [comment]
Perhaps, perhaps not. Who generates the URL (AJAX or not) for a button or link? Isn't that PF? or a combination of PF and Java?

There's the server's physical name.
There's the server's DNS alias.
And then the proxy server's name in front of both.

Obviously the proxy server has to deal with the first part of the URL.

Then there's the fact that the "app" is two WARs in an EAR running on WLS 12 with session sharing turned on (totally not spec) which to me really muddles things up.

**Comment by arnieHouston** [comment, marked as answer]
Turns out the proxy server configuration was wrong. Phew!
